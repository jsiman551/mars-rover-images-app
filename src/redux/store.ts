import { configureStore } from '@reduxjs/toolkit'
import photosDataReducer from './slices/photosData'
import filteringParamsReducer from './slices/filteringParams'

export const store = configureStore({
  reducer: {
    photosData: photosDataReducer,
    filteringParams: filteringParamsReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
