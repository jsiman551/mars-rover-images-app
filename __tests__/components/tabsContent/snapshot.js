import { render } from '@testing-library/react'
import TabsContent from '../../../src/components/tabsContent/index'
import { Provider } from 'react-redux'
import { store } from '@/redux/store'

it('renders tabs content unchanged', () => {
  const { container } = render(
    <Provider store={store}>
      <TabsContent />
    </Provider>,
  )
  expect(container).toMatchSnapshot()
})
