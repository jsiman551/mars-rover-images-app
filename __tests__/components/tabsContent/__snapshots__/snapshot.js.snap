// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tabs content unchanged 1`] = `
<div>
  <h2
    class="chakra-heading css-ft7kgn"
  >
    Mars Rover Photos
  </h2>
  <div
    class="chakra-tabs css-0"
    data-testid="tabs-container"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-1ev1cfq"
      role="tablist"
    >
      <button
        aria-controls="tabs-:r1:--tabpanel-0"
        aria-disabled="false"
        aria-selected="true"
        class="chakra-tabs__tab css-5wl5ud"
        data-index="0"
        id="tabs-:r1:--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        Curiosity
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel-1"
        aria-disabled="false"
        aria-selected="false"
        class="chakra-tabs__tab css-5wl5ud"
        data-index="1"
        id="tabs-:r1:--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Opportunity
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel-2"
        aria-disabled="false"
        aria-selected="false"
        class="chakra-tabs__tab css-5wl5ud"
        data-index="2"
        id="tabs-:r1:--tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Spirit
      </button>
    </div>
    <div
      class="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-:r1:--tab-0"
        class="chakra-tabs__tab-panel css-sjt5nk"
        id="tabs-:r1:--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="css-zb3t9w"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            Open Filters
          </button>
          <h2
            class="chakra-heading css-0"
          >
            Page: 1
          </h2>
        </div>
        <div
          class="css-xi606m"
          data-testid="loading-container"
        >
          <div
            class="chakra-spinner css-u4uvb9"
            data-testid="loading-spinner"
          >
            <span
              class="css-8b45rq"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-1"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r1:--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="css-zb3t9w"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            Open Filters
          </button>
          <h2
            class="chakra-heading css-0"
          >
            Page: 1
          </h2>
        </div>
        <div
          class="css-xi606m"
          data-testid="loading-container"
        >
          <div
            class="chakra-spinner css-u4uvb9"
            data-testid="loading-spinner"
          >
            <span
              class="css-8b45rq"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-2"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r1:--tabpanel-2"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="css-zb3t9w"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            Open Filters
          </button>
          <h2
            class="chakra-heading css-0"
          >
            Page: 1
          </h2>
        </div>
        <div
          class="css-xi606m"
          data-testid="loading-container"
        >
          <div
            class="chakra-spinner css-u4uvb9"
            data-testid="loading-spinner"
          >
            <span
              class="css-8b45rq"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-0"
    data-testid="sidebar-container"
  />
</div>
`;
